app/src/main/kotlin/at/bitfire/davdroid/resource/LocalAddressBookStore.kt

```diff
@@ -197,6 +194,42 @@ class LocalAddressBookStore @Inject constructor(
             accountManager.removeAccountExplicitly(addressBookAccount)
     }
 
+    /**
+     * Returns all address book accounts that belong to the given account.
+     *
+     * @param account    Account which has the address books.
+     * @return List of address book accounts.
+     */
+    fun getAddressBookAccounts(account: Account): List<Account> =
+        AccountManager.get(context).let { accountManager ->
+            accountManager.getAccountsByType(context.getString(R.string.account_type_address_book))
+                .filter { addressBookAccount ->
+                    account.name == accountManager.getUserData(
+                        addressBookAccount,
+                        LocalAddressBook.USER_DATA_ACCOUNT_NAME
+                    ) && account.type == accountManager.getUserData(
+                        addressBookAccount,
+                        LocalAddressBook.USER_DATA_ACCOUNT_TYPE
+                    )
+                }
+        }
+
+    /**
+     * Returns all address book accounts that belong to the given account in a flow.
+     *
+     * @param account    Account which has the address books.
+     * @return List of address book accounts as flow.
+     */
+    fun getAddressBookAccountsFlow(account: Account): Flow<List<Account>> = callbackFlow {
+        val accountManager = AccountManager.get(context)
+        val listener = OnAccountsUpdateListener { accounts ->
+            trySend(getAddressBookAccounts(account))
+        }
+        accountManager.addOnAccountsUpdatedListener(listener, null, true)
+        listener.onAccountsUpdated(accountManager.accounts)
```

Comment:
The `true` is for `updateImmediately`, which makes the `listener.onAccountsUpdated(accountManager.accounts)` unnecessary. (Please double-check)

To make this more clear, we could add the argument names for `null` and `true`.

Reply:
true!